File: 1-easy/02574-left-and-right-sum-differences/solution.cpp
class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        vector<int> res(nums.size(), 0);

        int sum = 0;
        for (int i{0}; i < nums.size(); ++i)  {
            res[i] += sum;
            sum += nums[i];
        }

        sum = 0;
        for (int i{(int) nums.size() - 1}; i >= 0; --i)  {
            res[i] = abs(res[i] - sum);
            sum += nums[i];
        }

        return res;
    }
};

File: 1-easy/02574-left-and-right-sum-differences/solution.rs
impl Solution {
    pub fn left_right_difference(nums: Vec<i32>) -> Vec<i32> {

        let mut res = vec![0_i32; nums.len()];

        let mut sum = 0_i32;
        for i in 0..nums.len() {
            res[i] += sum;
            sum += nums[i];
        }

        sum = 0;
        for i in (0..nums.len()).rev() {
            res[i] = (res[i] - sum).abs();
            sum += nums[i];
        }

        res
    }
}
